File: LeetCode/0643. Maximum Average Subarray I.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double tmp = 0;
        for(int i = 0; i < k; i++){
            tmp+=nums[i];
        }
        double ans = tmp/k;
        for(int i = k; i < nums.size(); i++){
            tmp-=nums[i-k];
            tmp+=nums[i];
            ans = max(ans, tmp/k);
        }
        return ans;
    }
};

File: LeetCode/0643. Maximum Average Subarray I.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        tmp = 0.0
        for i in range(k):
            tmp+=nums[i]
        ans = tmp/k
        for i in range(k, len(nums)):
            tmp-=nums[i-k]
            tmp+=nums[i]
            ans = max(ans, tmp/k)
        return ans
        